From 789c932421f501ba48c2dff6817f93de72f221fe Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Fri, 19 Nov 2021 20:21:12 +0100
Subject: [PATCH 3/6] Add systemd service file at expected location & unify
 naming

So it can be picked up by our recipes normally and will deploy the binary to the correct location as well and unify the naming so bitbake likes it more.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Pending
---
 ...service.in => luna-downloadmgr.service.in} |  6 ++---
 .../luna-downloadmgr.service}                 | 23 +++++++++++++++----
 2 files changed, 21 insertions(+), 8 deletions(-)
 rename files/launch/{luna-download-mgr.service.in => luna-downloadmgr.service.in} (82%)
 rename files/{launch/luna-download-mgr.sh.in => systemd/luna-downloadmgr.service} (61%)

diff --git a/files/launch/luna-download-mgr.service.in b/files/launch/luna-downloadmgr.service.in
similarity index 82%
rename from files/launch/luna-download-mgr.service.in
rename to files/launch/luna-downloadmgr.service.in
index ef1b9d4..49d4705 100644
--- a/files/launch/luna-download-mgr.service.in
+++ b/files/launch/luna-downloadmgr.service.in
@@ -24,8 +24,8 @@ After=ls-hubd.service
 [Service]
 Type=simple
 OOMScoreAdjust=-500
-EnvironmentFile=-/var/systemd/system/env/luna-download-mgr.env
+EnvironmentFile=-/var/systemd/system/env/luna-downloadmgr.env
 Environment=CHARSET=UTF-8
-ExecStartPre=/etc/systemd/system/scripts/luna-download-mgr.sh
-ExecStart=@WEBOS_INSTALL_BINDIR@/LunaDownloadMgr
+ExecStartPre=/etc/systemd/system/scripts/luna-downloadmgr.sh
+ExecStart=@WEBOS_INSTALL_SBINDIR@/LunaDownloadMgr
 Restart=on-failure
diff --git a/files/launch/luna-download-mgr.sh.in b/files/systemd/luna-downloadmgr.service
similarity index 61%
rename from files/launch/luna-download-mgr.sh.in
rename to files/systemd/luna-downloadmgr.service
index 0dc27a3..6c091c6 100644
--- a/files/launch/luna-download-mgr.sh.in
+++ b/files/systemd/luna-downloadmgr.service
@@ -1,4 +1,3 @@
-#!/bin/sh
 # Copyright (c) 2019 LG Electronics, Inc.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
@@ -15,7 +14,21 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-if [ -d /dev/cgroup ] ; then
-    mkdir -p /dev/cgroup/web
-    echo $$ > /dev/cgroup/web/tasks
-fi
+# The Download Manager Service
+
+[Unit]
+Description=default - "%n"
+Requires=ls-hubd.service
+After=ls-hubd.service
+
+[Service]
+Type=simple
+OOMScoreAdjust=-500
+EnvironmentFile=-/var/systemd/system/env/luna-downloadmgr.env
+Environment=CHARSET=UTF-8
+ExecStartPre=/etc/systemd/system/scripts/luna-downloadmgr.sh
+ExecStart=/usr/sbin/LunaDownloadMgr
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
-- 
2.37.3.windows.1

